src/nunavut/lang/py/__init__.py

```diff
@@ -332,3 +337,51 @@ def filter_longest_id_length(language: Language, attributes: typing.List[pydsdl.
         return max(map(len, map(functools.partial(filter_id, language), attributes)))
     else:
         return max(map(len, attributes))
+
+
+@template_language_filter(__name__)
+def filter_pickle(language: Language, instance: typing.Any) -> str:
+    pickled: str = base64.b85encode(gzip.compress(pickle.dumps(instance, protocol=4))).decode().strip()
+    segment_gen = map("".join, itertools.zip_longest(*([iter(pickled)] * 100), fillvalue=""))
+    return "\n".join(repr(x) for x in segment_gen)
+
+
+@template_language_filter(__name__)
+def filter_numpy_scalar_type(language: Language, t: pydsdl.Any) -> str:
+    def pick_width(w: int) -> int:
+        for o in [8, 16, 32, 64]:
+            if w <= o:
+                return o
+        raise ValueError(f"Invalid bit width: {w}")  # pragma: no cover
+
+    if isinstance(t, pydsdl.BooleanType):
+        return "bool"  # TODO: numpy.bool is deprecated in v1.20
+    if isinstance(t, pydsdl.SignedIntegerType):
+        return f"_np_.int{pick_width(t.bit_length)}"
+    if isinstance(t, pydsdl.UnsignedIntegerType):
+        return f"_np_.uint{pick_width(t.bit_length)}"
+    if isinstance(t, pydsdl.FloatType):
+        return f"_np_.float{pick_width(t.bit_length)}"
+    assert not isinstance(t, pydsdl.PrimitiveType), "Forgot to handle some primitive types"
+    return "object"  # TODO: numpy.object is deprecated in v1.20
```

Comment:
Indeed it is, this is why the original code uses `bool_` and `object_` instead of `bool` and `object`. What is the point of this change?

Also, the `_np_` prefix is missing for the boolean and primitive cases.

Reply:
Oh I see, I was checked out on an old commit: this was fixed in e0ec0fbf. I am updating everything now.